### PR TITLE
types(Extendable): add ButtonInteraction

### PR DESCRIPTION
**Pull request by ThatAnonyG**
**Please describe the changes this PR makes and why it should be merged:**
While working on a TypeScript project with the master branch you will get an error when trying to extend a ButtonInteraction structure. The ButtonInteraction class is being used in the `INTERACTION_CREATE` event handler as `Structure.get("ButtonInteraction");` which implies that it can be extended. So I added the ButtonInteraction class under the Extendable type. Also removed the MessageComponentInteraction from the JSDoc since it was not extendable either.

**Status and versioning classification:**
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

File: src/util/Structures.js [hunk 1]
```diff
@@ -20,7 +20,6 @@
  * * **`Role`**
  * * **`User`**
  * * **`CommandInteraction`**
- * * **`MessageComponentInteraction`**
```
Comment (SpaceEEC) [inline src/util/Structures.js:23]: You also need to remove this line then:
https://github.com/discordjs/discord.js/blob/a56fa6dd7b5386bc4cbc44bc4e88f109e3419bb5/src/util/Structures.js#L115

**Review by SpaceEEC — DISMISSED** [review]

File: src/util/Structures.js (hunk 1 above)
Comment (ThatAnonyG) [reply src/util/Structures.js:23]: It has been removed. Thanks for the review.

**Comment by monbrey** [comment]
imo `MessageComponentInteraction` should have been kept and added to the `Extendable` section, not removed.

**Comment by ThatAnonyG** [comment]
> imo `MessageComponentInteraction` should have been kept and added to the `Extendable` section, not removed.

It is not being used currently so I think it would cause some confusion for the end-users. The devs can add it back when they will actually use the extended MessageComponentInteraction classes.

**Comment by monbrey** [comment]
It is being used - ButtonInteraction and future component interactions will extend from it.

That makes it the logical structure to extend if you are adding properties that should exist on all component interactions.

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**